### PR TITLE
WIP: Switching Docs to `SortingAnalyzer`

### PR DESCRIPTION
**Pull request by zm711**
I switched postprocessing over to sorting_analyzer.compute notation.

I moved the WaveformExtractor down to a bottom legacy section and opened a blank `SortingAnalyzer` section to be filled in. 

I changed the high level quality metric over to the sorting_analyzer.

The low-level qc stuff should basically be the same right? I would just change that to a sorting_analyzer instead of the waveform_extractor, no?

### CODE REVIEW

**Comment by samuelgarcia** [comment]
Hi Zach.
Cool I will have a look very soon. Thanks a lot.

File: doc/modules/postprocessing.rst [hunk 1]
```diff
@@ -95,9 +86,9 @@ For dense waveforms, sparsity can also be passed as an argument.
 
 .. code-block:: python
 
-    pc = compute_principal_components(waveform_extractor=we,
-                                      n_components=3,
-                                      mode="by_channel_local")
+    sorting_analyzer.compute(input="principal_components",
```
Comment (zm711) [inline doc/modules/postprocessing.rst:89]:
```suggestion
    pc = sorting_analyzer.compute(input="principal_components",
```

File: doc/modules/postprocessing.rst [hunk 2]
```diff
@@ -112,7 +103,7 @@ and is not well suited for high-density probes.
 
 .. code-block:: python
 
-    similarity = compute_template_similarity(waveform_extractor=we, method='cosine_similarity')
+    sorting_analyzer.compute(input="template_similarity", method='cosine_similarity')
```
Comment (zm711) [inline doc/modules/postprocessing.rst:106]:
```suggestion
    similarity = sorting_analyzer.compute(input="template_similarity", method='cosine_similarity')
```

File: doc/modules/postprocessing.rst [hunk 3]
```diff
@@ -130,9 +121,9 @@ each spike.
 
 .. code-block:: python
 
-    amplitudes = computer_spike_amplitudes(waveform_extractor=we,
-                                           peak_sign="neg",
-                                           outputs="concatenated")
+    sorting_analyzer.compute(input="spike_amplitudes",
```
Comment (zm711) [inline doc/modules/postprocessing.rst:124]:
```suggestion
    amplitudes = sorting_analyzer.compute(input="spike_amplitudes",
```

File: doc/modules/postprocessing.rst [hunk 4]
```diff
@@ -150,15 +141,15 @@ with center of mass (:code:`method="center_of_mass"` - fast, but less accurate),
...
 .. code-block:: python
 
-    spike_locations = compute_spike_locations(waveform_extractor=we,
-                                              ms_before=0.5,
-                                              ms_after=0.5,
-                                              spike_retriever_kwargs=dict(
-                                                  channel_from_template=True,
-                                                  radius_um=50,
-                                                  peak_sign="neg"
+    sorting_analyzer.compute(input="spike_locations",
```
Comment (zm711) [inline doc/modules/postprocessing.rst:144]:
```suggestion
    spike_locations = sorting_analyzer.compute(input="spike_locations",
```

File: doc/modules/postprocessing.rst [hunk 5]
```diff
@@ -175,8 +166,7 @@ based on individual waveforms, it calculates at the unit level using templates.
 
 .. code-block:: python
 
-    unit_locations = compute_unit_locations(waveform_extractor=we,
-                                            method="monopolar_triangulation")
+    sorting_analyzer.compute(input="unit_locations", method="monopolar_triangulation")
```
Comment (zm711) [inline doc/modules/postprocessing.rst:169]:
```suggestion
    unit_locations = sorting_analyzer.compute(input="unit_locations", method="monopolar_triangulation")
```

File: doc/modules/postprocessing.rst [hunk 6]
```diff
@@ -219,10 +209,10 @@ with shape (num_units, num_units, num_bins) with all correlograms for each pair
 
 .. code-block:: python
 
-    ccgs, bins = compute_correlograms(waveform_or_sorting_extractor=we,
-                                      window_ms=50.0,
-                                      bin_ms=1.0,
-                                      method="auto")
+    sorting_analyer.compute(input="correlograms",
```
Comment (zm711) [inline doc/modules/postprocessing.rst:212]:
```suggestion
    ccg = sorting_analyzer.compute(input="correlograms",
```

File: doc/modules/postprocessing.rst [hunk 7]
```diff
@@ -236,10 +226,10 @@ This extension computes the histograms of inter-spike-intervals. The computed ou
 
 .. code-block:: python
 
-    isi_histogram, bins =  compute_isi_histograms(waveform_or_sorting_extractor=we,
-                                                  window_ms=50.0,
-                                                  bin_ms=1.0,
-                                                  method="auto")
+    sorting_analyer.compute(input="isi_histograms"
```
Comment (zm711) [inline doc/modules/postprocessing.rst:229]:
```suggestion
   isi =  sorting_analyer.compute(input="isi_histograms"
```

File: doc/modules/qualitymetrics.rst [hunk 8]
```diff
@@ -48,17 +48,19 @@ This code snippet shows how to compute quality metrics (with or without principa
...
 .. code-block:: python
 
-    we = si.load_waveforms(folder='waveforms') # start from a waveform extractor
+    sorting_analyzer = si.load_sorting_analyzer(folder='waveforms') # start from a sorting_analyzer
 
-    # without PC
-    metrics = compute_quality_metrics(waveform_extractor=we, metric_names=['snr'])
+    # without PC (depends on "waveforms", "templates", and "noise_levels")
+    sorting_analyzer.compute(input="quality_metrics", metric_names=['snr'], skip_pc_metrics=False)
+    metrics = sorting_analyzer.get_extension(extension_name="quality_metrics")
```
Comment (zm711) [inline doc/modules/qualitymetrics.rst:55]:
```suggestion
    # without PC (depends on "waveforms", "templates", and "noise_levels")
    qm_ext = sorting_analyzer.compute(input="quality_metrics", metric_names=['snr'], skip_pc_metrics=True)
    metrics = qm_ext.get_data()
```

**Review by zm711 — COMMENTED** [review]
I thought that extensions were no longer returned after computation, but during my testing I realized they were. So I'm just going to commit those fixes here.

**Comment by samuelgarcia** [comment]
Hi Zach.
I will merge this now.
And start another PR to continue the work.

**Pull request merged**